$DragonFly: src/gnu/usr.bin/cc3/cc_int/patches/Attic/optabs.c.patch,v 1.1 2004/03/05 21:38:44 joerg Exp $
Index: optabs.c
===================================================================
RCS file: /cvs/src/contrib/gcc-3.3/gcc/optabs.c,v
retrieving revision 1.1
diff -u -r1.1 optabs.c
--- optabs.c	2 Feb 2004 23:09:49 -0000	1.1
+++ optabs.c	4 Mar 2004 15:18:42 -0000
@@ -703,6 +703,26 @@
   if (target)
     target = protect_from_queue (target, 1);
 
+  if (flag_propolice_protection
+      && binoptab->code == PLUS
+      && op0 == virtual_stack_vars_rtx
+      && GET_CODE(op1) == CONST_INT)
+    {
+      int icode = (int) binoptab->handlers[(int) mode].insn_code;
+      if (target)
+	temp = target;
+      else
+	temp = gen_reg_rtx (mode);
+
+      if (! (*insn_data[icode].operand[0].predicate) (temp, mode)
+	  || GET_CODE (temp) != REG)
+	temp = gen_reg_rtx (mode);
+
+      emit_insn (gen_rtx_SET (VOIDmode, temp,
+			      gen_rtx_PLUS (GET_MODE (op0), op0, op1)));
+      return temp;
+    }
+
   if (flag_force_mem)
     {
       op0 = force_not_mem (op0);
